set case properly for sqllite testing and any future sql

skandragon opened the pull request:
PostgreSQL is also case sensitive, so should that end up used (which we do use) this would break there as well.

It's strange that string comparisons in MySQL are case insensitive by default coming from a PostgreSQL heavy background.

billiford commented:
LGTM. Looks like our linter needs to be updated, working on that now.

billiford commented:
@skandragon I think we should do something like this.
```sql
AND UPPER(kind) in (UPPER('Deployment'), UPPER('StatefulSet'), UPPER('ReplicaSet'), UPPER('Ingress'), UPPER('Service'), UPPER('DaemonSet'))
```
This will make the select case-insensitive regardless of the data-store type.

billiford commented on internal/sql/client.go:
```diff
@@ -256,7 +256,7 @@ func (c *client) GetKubernetesProviderAndPermissions(name string) (kubernetes.Pr
 	var rs []kubernetes.Resource
 	db := c.db.Select("account_name, cluster").
-		Where("spinnaker_app = ? AND kind in ('deployment', 'statefulSet', 'replicaSet', 'ingress', 'service', 'daemonSet')",
+		Where("spinnaker_app = ? AND kind in ('Deployment', 'StatefulSet', 'ReplicaSet', 'Ingress', 'Service', 'DaemonSet')",
```
I think both in operations should be
```sql
UPPER(kind) in (UPPER('Deployment'), UPPER('StatefulSet'), UPPER('ReplicaSet'), UPPER('Ingress'), UPPER('Service'), UPPER('DaemonSet'))
```
Will need to update the tests for these as well.

skandragon replied:
That would work as well I think.  I'll make the change and add a tests.

skandragon commented:
I changed this slightly from your example, as passing `UPPER()` a literal seems wasteful.  I just changed the values to all uppercase, and only use `UPPER(kind)`.  Also, tests are updated.
